### DISCUSSION TITLE
New Jetpack CameraX library: Camera Viewfinder Compose

### DISCUSSION THREAD

**Discussion by vanshg** [ujizin/Camposer#46] [Ideas]
At Google I/O 2024, Google announced a new Jetpack CameraX library: `camera-viewfindfer-compose`. It is currently only at version `alpha01`.

https://developer.android.com/jetpack/androidx/releases/camera#viewfinder-compose-1.0.0-alpha01

Eventually,  it _may_ (or may not) make sense to incorporate into this project. However, as the name implies, the viewfinder is just that. It is not "batteries included" like Camposer is.

**Comment by ujizin** [comment]
Hey @vanshg, sorry for the delayed response.

Thanks for sharing the library; it’s a really interesting one. I’ll look into the possibility of implementing it in Camposer once it becomes more stable, if feasible 😃

**Comment by ujizin** [comment]
It seems that Google has released a Camera-Compose library as well:
https://android-developers.googleblog.com/2024/12/whats-new-in-camerax-140-and-jetpack-compose-support.html

It will likely adopt the "batteries-included" approach seen in Camposer. 

In the future, it might no longer make sense to include this functionality in Camposer, as both serve the same purpose, and the Google library is official
